product_code/user_service/api/src/main/scala/de/upb/cs/uc4/user/api/UserService.scala

```diff
@@ -31,28 +31,31 @@ trait UserService extends UC4Service {
   def getUser(username: String): ServiceCall[NotUsed, User]
 
   /** Get all users from the database */
-  def getAllUsers(usernames: Option[String]): ServiceCall[NotUsed, GetAllUsersResponse]
+  def getAllUsers(usernames: Option[String], only_active: Option[Boolean]): ServiceCall[NotUsed, GetAllUsersResponse]
 
   /** Adds the contents of the postMessageUser, authUser to authentication users and user to users */
   def addUser(): ServiceCall[PostMessageUser, User]
 
   /** Update an existing user */
   def updateUser(username: String): ServiceCall[User, Done]
 
-  /** Delete a user from the database */
-  def deleteUser(username: String): ServiceCall[NotUsed, Done]
+  /** Completely deletes a users from the database */
+  def forceDeleteUser(username: String): ServiceCall[NotUsed, Done]
+
+  /** Flags a user as deleted and deletes (not required) personal info */
+  def softDeleteUser(username: String): ServiceCall[NotUsed, Done]
 
   // STUDENT
   /** Get all students from the database */
-  def getAllStudents(usernames: Option[String]): ServiceCall[NotUsed, Seq[Student]]
+  def getAllStudents(usernames: Option[String], only_active: Option[Boolean]): ServiceCall[NotUsed, Seq[Student]]
```

Comment:
```suggestion
  def getAllStudents(usernames: Option[String], onlyActive: Option[Boolean]): ServiceCall[NotUsed, Seq[Student]]
```
We are using camelCase